rocksdb_replicator/replicated_db.cpp

```diff
@@ -342,10 +337,11 @@ void RocksDBReplicator::ReplicatedDB::pullFromUpstream() {
             // no updates consecutively, and the upstream says it's NOT a leader.
             // Therefore we reset upstream.
             db->pullFromUpstreamNoUpdates_++;
-            if (response.role != ReplicaRole::LEADER
+            if (response.__isset.role && response.role != ReplicaRole::LEADER
```

Comment:
thanks for fixing this. otherwise, it will throw NPE

Reply:
I don't think it will throw NPE, it'll just be un-initialized, in which case it'll be `NOOP`

Reply:
OK, it will `un-initialized`, not NPE in c++. if the logic is in Java, if will be NPE.